java-cfenv-boot/src/main/java/io/pivotal/cfenv/spring/boot/GenAIEmbeddingCfEnvProcessor.java

```diff
@@ -46,9 +51,17 @@ public boolean accept(CfService service) {
 	public void process(CfCredentials cfCredentials, Map<String, Object> properties) {
 		properties.put("spring.ai.openai.api-key", "redundant");
 
-		properties.put("spring.ai.openai.embedding.base-url", cfCredentials.getString("api_base"));
-		properties.put("spring.ai.openai.embedding.api-key", cfCredentials.getString("api_key"));
-		properties.put("spring.ai.openai.embedding.options.model", cfCredentials.getString("model_name"));
+		boolean endpointFormat = cfCredentials.getMap().containsKey("endpoint");
+		if (endpointFormat) {
+			Map<String, String> endpoint = (Map<String, String>)cfCredentials.getMap().get("endpoint");
+			properties.put("spring.ai.openai.embedding.base-url", endpoint.get("api_base") + "/openai");
+			properties.put("spring.ai.openai.embedding.api-key", endpoint.get("api_key"));
+			properties.put("spring.ai.openai.embedding.options.model", "tanzu://embedding");
```

Comment:
Same note as above, this will always try to configure an embedding model, even when there is none.

Reply:
Same as comment above